libs/components/code-examples/src/lib/modules/filter-bar/filter-bar/basic/example.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component } from '@angular/core';
+import {
+  SkyFilterBarFilterItem,
+  SkyFilterBarFilterModalConfig,
+  SkyFilterBarFilterValue,
+  SkyFilterBarModule,
+} from '@skyux/filter-bar';
+
+import { FilterModalComponent } from './filter-modal.component';
+import { FILTER_SELECTION_VALUES } from './filter-selection-values';
+
+/**
+ * @title Filter bar basic example
+ */
+@Component({
+  selector: 'app-filter-bar-basic-example',
+  imports: [SkyFilterBarModule],
+  templateUrl: './example.component.html',
+})
+export class FilterBarBasicExampleComponent {
+  public appliedFilters: SkyFilterBarFilterItem[] | undefined;
+  public selectedFilterIds: string[] | undefined;
+
+  protected communityConnectionConfig: SkyFilterBarFilterModalConfig = {
+    modalComponent: FilterModalComponent,
+    modalSize: 'medium',
+    additionalContext: this.#getFilterContext('community-connection'),
+  };
+  protected currentGradeConfig: SkyFilterBarFilterModalConfig = {
+    modalComponent: FilterModalComponent,
+    modalSize: 'medium',
+    additionalContext: this.#getFilterContext('current-grade'),
+  };
+  protected enteringGradeConfig: SkyFilterBarFilterModalConfig = {
+    modalComponent: FilterModalComponent,
+    modalSize: 'medium',
+    additionalContext: this.#getFilterContext('entering-grade'),
+  };
+  protected roleConfig: SkyFilterBarFilterModalConfig = {
+    modalComponent: FilterModalComponent,
+    modalSize: 'medium',
+    additionalContext: this.#getFilterContext('role'),
+  };
+  protected staffAssignedConfig: SkyFilterBarFilterModalConfig = {
+    modalComponent: FilterModalComponent,
```

Comment:
I realize having one component for all the filter modals is efficient, but we don't want to communicate to the consumer (via code example) that all filters should use only one modal and pass "additional context" to differentiate. For the sake of communicating that a filter button _can_ have a dedicated modal component, I would create a separate component for each filter (the consumer can then decide if using one component for multiple filters is a good idea for their use case); it's still appropriate to load the data through the async mechanism described below, however.